### PR TITLE
Make build environment manylinux2014 compatible

### PR DESCRIPTION
**Pull request by nitins17**
Modifies the build environment to be manylinux2014 compatible

Changes include:
* Added devtoolset-8 to the Dockerfile
* Created a config option for manylinux2014 in {cpu,gpu}.bazelrc
* Added manylinux2014 compatible glibc and libstdc++ libraries in devtoolset-8 (/dt8)

### CODE REVIEW

File: tf_sig_build_dockerfiles/devel.usertools/gpu.bazelrc [hunk 1]
```diff
@@ -37,6 +37,10 @@ build --action_env=GCC_HOST_COMPILER_PATH="/dt7/usr/bin/gcc"
 build --action_env=LD_LIBRARY_PATH="/usr/local/cuda:/usr/local/cuda/lib64:/usr/local/cuda/extras/CUPTI/lib64:/usr/local/tensorrt/lib"
 build --crosstool_top=@ubuntu18.04-gcc7_manylinux2010-cuda11.2-cudnn8.1-tensorrt7.2_config_cuda//crosstool:toolchain
 
+# Set up a manylinux2014 config option
```
Comment (angerson) [inline tf_sig_build_dockerfiles/devel.usertools/gpu.bazelrc:40]: I'd also suggest that you drop the option here and just make manylinux2014 the default while you're using this PR. It will be easier to test the WIP containers that way, because they'll be the exact same as the regular ones and you won't need to change any flags outside.

We can revisit the preferred method of integration once you've verified everything works and you're ready to merge these -- which I expect won't happen for a while, until you've done more testing.

File: tf_sig_build_dockerfiles/builder.devtoolset/build_devtoolset.sh [hunk 2]
```diff
@@ -15,7 +15,7 @@
 # ==============================================================================
 #
 # Builds a devtoolset cross-compiler targeting manylinux 2010 (glibc 2.12 /
-# libstdc++ 4.4).
+# libstdc++ 4.4) or manylinux2014 (glibc 2.17 / libstdc++ 4.8).
```
Comment (angerson) [inline tf_sig_build_dockerfiles/builder.devtoolset/build_devtoolset.sh:18]: Is this file pulled directly from tensorflow's master branch? If so, you can also make a separate PR to update it to match -- it should be kept up to date with TensorFlow's (eventually, it should only live here, but for now it should be mirrored).

**Comment by angerson** [comment]
You should be able to apply that label whenever you like, as a TF org member.

File: tf_sig_build_dockerfiles/builder.devtoolset/build_devtoolset.sh (hunk 2 above)
Comment (nitins17) [reply tf_sig_build_dockerfiles/builder.devtoolset/build_devtoolset.sh:18]: The only difference this file has with tensorflow's master branch is [e2d2d48](https://github.com/tensorflow/build/pull/52/commits/e2d2d48a1cd0a29dab87ba7900a05cb69e57715c) where I removed Python 3.6.

File: tf_sig_build_dockerfiles/devel.usertools/gpu.bazelrc (hunk 1 above)
Comment (nitins17) [reply tf_sig_build_dockerfiles/devel.usertools/gpu.bazelrc:40]: Oh sure, I'll push a commit to update this. Thanks!

File: tf_sig_build_dockerfiles/builder.devtoolset/build_devtoolset.sh (hunk 2 above)
Comment (nitins17) [reply tf_sig_build_dockerfiles/builder.devtoolset/build_devtoolset.sh:18]: I'll push commit to make it mirror the master branch's file.

**Comment by angerson** [comment]
Huh. I guess the push didn't work because

> Secrets are not passed to workflows that are triggered by a pull request from a fork.

So what I think you'll have to do instead is create a branch on this repository directly (you should have permissions to do this) and then create a PR based on that branch, instead of a PR from your fork.

**Comment by nitins17** [comment]
Closed this as #57 tracks the manylinux2014 upgrade.
